utils/prof_utils.py

```diff
@@ -0,0 +1,523 @@
+#!/usr/bin/env python3
+
+import numpy as np
+import re
+import sys
+import sys
+import logging
+import argparse
+from scipy.interpolate import UnivariateSpline
+import matplotlib
+matplotlib.use('Agg')
+import matplotlib.pyplot as plt
+from astropy.time import Time
+
+from stickel import Stickel
+
+logger = logging.getLogger(__name__)
+
+def get_from_bestprof(file_loc):
+    """
+    Get info from a bestprof file
+
+    Parameters:
+    -----------
+    file_loc: string
+        The path to the bestprof file
+    
+    Returns:
+    --------
+    [obsid, pulsar, dm, period, period_uncer, obsstart, obslength, profile, bin_num]: list
+        obsid: int
+            The observation ID
+        pulsar: string
+            The name of the pulsar
+        dm: float
+            The dispersion measure of the pulsar
+        period: float
+            The period of the pulsar
+        period_uncer: float
+            The uncertainty in the period measurement
+        obsstart: int
+            The beginning time of the observation
+        obslength: float
+            The length of the observation in seconds
+        profile: list
+            A list of floats containing the profile data
+        bin_num: int
+            The number of bins in the profile
+    """
+
+    with open(file_loc,"r") as bestprof:
+        lines = bestprof.readlines()
+        # Find the obsid by finding a 10 digit int in the file name
+        obsid = re.findall(r'(\d{10})', lines[0])[0]
+        try:
+            obsid = int(obsid)
+        except:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [No exception type(s) specified](https://app.codacy.com/app/CIRA-Pulsars-and-Transients-Group/vcstools/pullRequest?prid=4645369)